pkg/controller/wip/controller_test.go

```diff
@@ -523,6 +683,123 @@ func TestReconcileBindingWithParameters(t *testing.T) {
 	}
 }
 
+func TestReconcileBindingDelete(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, _, _, fakeBindingClient, testController, sharedInformers, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	fakeBindingClient.Bindings = map[string]struct{}{
+		fmt.Sprintf("%s:%s", instanceGUID, bindingGUID): {},
+	}
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := &v1alpha1.Instance{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "test-instance",
+			Namespace: "test-ns",
+		},
+		Spec: v1alpha1.InstanceSpec{
+			ServiceClassName: "test-serviceclass",
+			PlanName:         "default",
+			OSBGUID:          instanceGUID,
+		},
+	}
+
+	sharedInformers.Instances().Informer().GetStore().Add(instance)
+
+	binding := &v1alpha1.Binding{
+		ObjectMeta: v1.ObjectMeta{
+			Name:              "test-binding",
+			Namespace:         "test-ns",
+			DeletionTimestamp: &metav1.Time{},
+			Finalizers:        []string{"kubernetes"},
+		},
+		Spec: v1alpha1.BindingSpec{
+			InstanceRef: v1.ObjectReference{Name: "test-instance", Namespace: "test-ns"},
+			OSBGUID:     bindingGUID,
+			SecretName:  "test-secret",
+		},
+	}
+
+	testController.reconcileBinding(binding)
+
+	kubeActions := fakeKubeClient.Actions()
+	// The two actions should be:
+	// 0. Getting the secret
+	// 1. Deleting the secret
+	if e, a := 2, len(kubeActions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
```

Comment:
I'd like to argue against this pattern here and in all other invocations in this PR. It adds two new variables -- and their associated complexity -- for the reader to understand, and it looks like the payoff is so that you don't have to re-type the values in the logging statements that lie inside the `if` block.

If you disagree, please continue with this PR. I want to voice my opinion here and avoid a bikeshedding session herein.